### PR TITLE
Add support to Unsigned Integer data type.

### PR DESCRIPTION
**Pull request by sbernard31**
I need to validate IP for ULong from https://github.com/jOOQ/jOOU before to integrate this.

(CQ : https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22566)

### CODE REVIEW

File: leshan-core/src/main/java/org/eclipse/leshan/core/util/datatype/NumberUtil.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                throw new IllegalStateException(
+                        String.format("%s  : can not convert negative number to an unsigned long", number));
+            }
+            return ULong.valueOf(longValue);
+        }
+        if (number instanceof BigInteger) {
+            // check big integer is not too big for a long
+            BigInteger bigInt = (BigInteger) number;
+            if (bigInt.signum() == -1 || bigInt.compareTo(ULong.MAX_VALUE) > 0) {
+                throw new IllegalStateException(String.format("%s  : can not be store in an unsigned  long", bigInt));
```
Comment (dachaac) [inline leshan-core/src/main/java/org/eclipse/leshan/core/util/datatype/NumberUtil.java:83]: double space in "unsigned  long"

**Comment by sbernard31** [comment]
@dachaac :pray:  thx you very much for all the reviews !

**Comment by sbernard31** [comment]
I'm still waiting  about CQ for this one :/

**Review by msangoi — APPROVED** [review]

**Comment by sbernard31** [comment]
@boaks, @sophokles73  about CQ/IP do you know if code could be integrated in master or even in milestone release without CQ approval ?
I mean : Is CQ only needed for release ?
(if you don't know to not start research for me, I will search on my side)

**Comment by boaks** [comment]
In the master-branch, no problem. 
For a milestone, I don't know. 
I would argument, that it depends more on your own check of the license of that material. If all is well, I would also go for a milestone. Sometimes a comment on the CQ, that you want to release something, helps also to speed up. But not always.

**Comment by sbernard31** [comment]
I finally integrated this in `master`.
This is too long to get CQ validation or even to get answer to questions.
So let's bet this will be accepted. (other version of this library is already accepted. I think the issue here could be mainly because we modified a bit the code)

**Pull request merged**